base/utf16.jl

```diff
@@ -143,3 +143,24 @@ function utf16(p::Union{Ptr{UInt16}, Ptr{Int16}})
     while unsafe_load(p, len+1) != 0; len += 1; end
     utf16(p, len)
 end
+
+function map(fun, str::UTF16String)
+    buf = UInt16[]
+    sizehint!(buf, length(str.data))
+    for ch in str
+        c2 = fun(ch)
+        !isa(c2, Char) &&
```

Comment:
Does this structure result in coverage reporting separate hits for both lines? I don't see a test that hits this error path.